src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusCollectionManager.cs

```diff
@@ -190,7 +190,7 @@ private bool ExecuteCollect(bool openMetricsRequested)
     {
         this.exporter.OnExport = this.onCollectRef;
         this.exporter.OpenMetricsRequested = openMetricsRequested;
-        var result = this.exporter.Collect(Timeout.Infinite);
+        var result = this.exporter.Collect!(Timeout.Infinite);
```

Comment:
I don't know enough about Prometheus, so I may need some help on this line.

What assurances do we have that the `Collect` func is not null here?

Reply:
I think it is OK. The code as it is today is essentially doing the `!`. I did add a `Debug.Assert` though to (hopefully) catch any future/possible dev mistakes. IMO always good to have a `Debug.Assert` anywhere we have a `!` 😄